test/parallel/test-net-error-twice.js

```diff
@@ -1,33 +1,37 @@
 'use strict';
-var common = require('../common');
-var assert = require('assert');
-var net = require('net');
+const common = require('../common');
+const assert = require('assert');
+const net = require('net');
 
-var buf = new Buffer(10 * 1024 * 1024);
+const buf = new Buffer(10 * 1024 * 1024);
 
 buf.fill(0x62);
 
 var errs = [];
 
-var srv = net.createServer(function onConnection(conn) {
-  conn.write(buf);
+const srv = net.createServer(function onConnection(conn) {
+  if (common.isWindows) {
+    // Windows-specific handling. See:
+    //   * https://github.com/nodejs/node/pull/4062
+    //   * https://github.com/nodejs/node/issues/4057
+    setTimeout(() => { conn.write(buf); }, 100);
```

Comment:
@bnoordhuis Concise-but-naive answer: Without the delay, the test is flaky on Windows 2012 r2 on the CI infrastructure.

Less concise and marginally less naive: In this test, `conn.write(buf)` is suppose to get an error (either `EPIPE` or `ECONNRESET` depending on timing). Unfortunately, on Windows 2012 r2 only, it appears that the call to `client.destroy()` often returns before the socket is set to throw errors. On Windows 2012 r2 only, the `conn.write()` would not trigger an error resulting in `assert.equal(errs.length, 1);` firing (because `errs.length` is 0). I tried `setImmediate()` and that may have improved things but it did not cure the flakiness. Using the timeout cured the flakiness. (I did not experiment with a timeout of less than 100 milliseconds.)

As for the cause of this on Windows 2012 r2, I don't know and would certainly be receptive to some guidance on figuring it out (or someone simply doing the work and identifying the source of the issue). Operating system limitation? Bug in libuv? Bug in `socket.destroy`? I don't know.